lua/paramset.lua

```diff
@@ -39,12 +39,46 @@ function ParamSet:add_separator()
 end
 
 --- add generic parameter
+-- helper function to add param to paramset
+-- two uses:
+-- - pass "param" table with optional "action" function
+-- - pass keyed table to generate "param" table. required keys are "type" and "id"
 function ParamSet:add(args)
-  local param = args.param -- param is mandatory
+  local param = args.param
+  if param == nil then
+    if args.type == nil then
+      print("paramset.add() error: type required")
+      return nil
+    elseif args.id == nil then
+      print("paramset.add() error: id required")
+      return nil
+    end
+
+    local id = args.id
+    local name = args.name or id
+
+    if args.number then
```

Comment:
hmm, but shouldn't we check args.type here instead?

Reply:
sorry totally messed this up